dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/BasicTools.java

```diff
@@ -27,20 +27,53 @@
 public class BasicTools {
 
     @Tool(name = "mixedContentTool", title = "Mixed Content Tool", description = "Returns Text, Audio or Image Content List")
-    public ToolResponse mixedContentTool(@ToolArg(name = "input", description = "input to echo") String input) {
-        TextContent text = new TextContent("Echo: " + input);
-
-        ImageContent image = new ImageContent(
-                                              "base64-encoded-image",
-                                              "image/png",
-                                              null);
-
-        AudioContent audio = new AudioContent(
-                                              "base64-encoded-audio",
-                                              "audio/mpeg",
-                                              null);
-
-        return ToolResponse.success(List.of(text, image, audio));
+    public List<Content> mixedContentTool(@ToolArg(name = "input", description = "input to echo") String input) {
+        return List.of(
+                       new TextContent("Echo: " + input),
+
+                       new ImageContent(
+                                        "base64-encoded-image",
+                                        "image/png",
+                                        null),
+
+                       new AudioContent(
+                                        "base64-encoded-audio",
+                                        "audio/mpeg",
+                                        null));
+    }
+
+    @Tool(
+          name = "textContentListTool",
+          title = "Text Content List Tool",
+          description = "Returns a list of content objects")
+    public List<Content> textContentListTool(
+                                             @ToolArg(name = "input", description = "input string to echo back as content") String input) {
+        return List.of(new TextContent("Echo: " + input));
+    }
+
+    @Tool(
+          name = "imageContentListTool",
+          title = "Image Content List Tool",
+          description = "Returns a list of content objects")
+    public List<Content> imageContentListTool(
+                                              @ToolArg(name = "imageData", description = "Base64-encoded image") String imageData) {
+        return List.of(new ImageContent(
+                                        imageData,
+                                        "image/png",
+                                        null));
+    }
+
+    @Tool(
+          name = "audioContentListTool",
+          title = "Audio Content List Tool",
+          description = "Returns a list of content objects")
+    public List<Content> audioContentListTool(
```

Comment:
You've got lots of tests for returning `List<Content>`, but we also need the tool method to be able to return `TextContent`, `AudioContent` or `ImageContent` directly.